cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -49,25 +61,73 @@
  */
 public class CLIMain {
 
-  private final CLI cli;
+  public static final String NAME_VERBOSE = "verbose";
+  public static final String NAME_URI = "uri";
+  public static final String NAME_VERIFY_SSL = "verify_ssl";
+  public static final String NAME_AUTOCONNECT = "autoconnect";
+  public static final String NAME_NAMESPACE = "namespace";
+
+  private static final boolean DEFAULT_VERIFY_SSL = true;
+  private static final boolean DEFAULT_AUTOCONNECT = true;
+  private static final boolean DEFAULT_VERBOSE = false;
+  private static final String DEFAULT_NAMESPACE = Constants.DEFAULT_NAMESPACE;
+
+  private static final Option URI_OPTION = new Option(
+    "u", "uri", true, "CDAP instance URI to interact with in" +
+    " the format \"[http[s]://]<hostname>[:<port>[/<namespace>]]\"." +
+    " Defaults to \"" + getDefaultURI() + "\".");
+
+  private static final Option NAMESPACE_OPTION = new Option(
+    "n", "namespace", true, "The CDAP namespace to use." +
+    " Defaults to \"" + DEFAULT_NAMESPACE + "\".");
 
+  private static final Option VERIFY_SSL_OPTION = new Option(
+    "s", "verify-ssl", true, "If \"true\", verify SSL certificate when making requests." +
+    " Defaults to \"" + DEFAULT_VERIFY_SSL + "\".");
+
+  private static final Option AUTOCONNECT_OPTION = new Option(
+    "a", "autoconnect", true, "If \"true\", try provided connection" +
+    " (from " + URI_OPTION.getLongOpt() + ")" +
+    " upon launch or try default connection if none provided." +
+    " Defaults to \"" + DEFAULT_AUTOCONNECT + "\".");
+
+  private static final Option VERBOSE_OPTION = new Option(
+    "v", "verbose", true, "If \"true\", print all exception stack traces." +
+    " Defaults to \"" + DEFAULT_VERBOSE + "\".");
+
+  private final CLI cli;
   private final Iterable<CommandSet<Command>> commands;
 
-  public CLIMain(final CLIConfig cliConfig) throws URISyntaxException, IOException {
-    Injector injector = Guice.createInjector(
-      new AbstractModule() {
-        @Override
-        protected void configure() {
-          bind(CLIConfig.class).toInstance(cliConfig);
-          bind(ClientConfig.class).toInstance(cliConfig.getClientConfig());
-          bind(CConfiguration.class).toInstance(CConfiguration.create());
+  /**
+   * @param output output to print to
+   * @param namespace the CDAP namespace to use initially
+   * @param uri provided URI of CDAP instance
+   * @param autoconnect if true, try provided connection (or default from CConfiguration) before startup
+   * @param verbose if true, log all exception stack traces
+   * @throws URISyntaxException
+   * @throws IOException
+   */
+  @Inject
+  public CLIMain(Injector injector, PrintStream output,
```

Comment:
Injecting an ``Injector`` doesn't seem like a common pattern, right? Was there a problem with the old method of using ``Guice.createInjector``?